src/Config.ts

```diff
@@ -185,9 +189,17 @@ export interface IConfigDatastore {
 
 export interface IConfigRoomRule {
     /**
-     * Room ID or alias
+     * Room ID or alias. Optional.
+     *
+     * Either `room` or `roomRegex` must be provided.
+     */
+    room?: string;
+    /**
+     * A regex to match a Room ID or alias. Optional.
+     *
+     * Either `room` or `roomRegex` must be provided.
      */
-    room: string;
+    roomRegex?: string;
```

Comment:
I recommend converting the strings to RegExp here.
A bit for performance reasons, but mainly to catch config errors (alias invalid RegExp strings) at startup.

`new RegExp(string)`